types/sequelize-cursor-pagination/index.d.ts

```diff
@@ -0,0 +1,61 @@
+// Type definitions for sequelize-cursor-pagination 1.1
+// Project: https://github.com/Kaltsoon/sequelize-cursor-pagination
+// Definitions by: pilagod <https://github.com/pilagod>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped/
+// TypeScript Version: 2.9
+
+import Sequelize = require('sequelize');
+
+declare namespace SequelizeCursorPagination {
+    interface Cursors {
+        before: string | null;
+        after: string | null;
+        hasNext: boolean;
+        hasPrevious: boolean;
+    }
+
+    interface BasicPaginateOptions<T> {
+        where?: Sequelize.WhereOptions<T> | Sequelize.where | Sequelize.fn | Array<Sequelize.col | Sequelize.and | Sequelize.or | string>;
+        attributes?: Sequelize.FindOptionsAttributesArray | { include?: Sequelize.FindOptionsAttributesArray, exclude?: string[] };
+        include?: Array<Sequelize.Model<any, any> | Sequelize.IncludeOptions>;
+        limit?: number; // limit the number of records returned
+        desc?: boolean; // [default: false]
+        before?: string; // the before cursor
+        after?: string; // the after cursor
+        paginationField?: string; // [default: primaryKeyField]
+    }
+
+    interface PaginateRawOptions<T> extends BasicPaginateOptions<T> {
+        raw: true;
+    }
+
+    interface PaginateInstOptions<T> extends BasicPaginateOptions<T> {
+        raw?: boolean;
+    }
+
+    interface WithPaginationOptions<MethodName extends string> {
+        methodName?: MethodName; // [default: 'paginate']
+        primaryKeyField?: string; // [default: 'id']
+    }
+
+    type WithPaginationModel<MethodName extends string, Model, ModelInst, ModelAttrs> = Model & (
+        {
```

Comment:
Can't you avoid the intersection and just write this as a type with two overloads?

```
{
  [key in MethodName]: {
    (options?: PaginateRawOptions<ModelAttrs>): { results: ModelAttrs[], cursors: Cursors }
    (options?: PaginateInstOptions<ModelAttrs>): { results: ModelInst[], cursors: Cursors }
  }
}
```

Reply:
I didn't realize this refactoring ! This makes the type more clear, many thanks 😃